app/controllers/predicates/TaxYearAction.scala

```diff
@@ -38,25 +38,23 @@ class TaxYearAction @Inject()(taxYear: Int, missingTaxYearReset: Boolean)(
 
   override def refine[A](request: User[A]): Future[Either[Result, User[A]]] = {
     implicit val implicitUser: User[A] = request
-
+    val sameTaxYear = request.session.get(TAX_YEAR).exists(_.toInt == taxYear)
     Future.successful(
-      if (!appConfig.taxYearErrorFeature || taxYear == appConfig.defaultTaxYear) {
-        val sameTaxYear = request.session.get(TAX_YEAR).exists(_.toInt == taxYear)
-
-        if (sameTaxYear || !missingTaxYearReset) {
-          Right(request)
-        } else {
-          logger.info("[TaxYearAction][refine] Tax year provided is different than that in session. Redirecting to overview.")
-          Left(
-            Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))
-              .addingToSession(TAX_YEAR -> taxYear.toString)
-          )
-        }
+      //      if (!appConfig.taxYearErrorFeature || taxYear == appConfig.defaultTaxYear) {
```

Comment:
I think you will need to revert this change. I think the commented sections should not be commented. Please make sure the class is same (or does the same thing) as the TaxYearAction from  personal-income-tax-submission-frontend